beacon_node/beacon_chain/src/sync_committee_rewards.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{BeaconChain, BeaconChainError, BeaconChainTypes};
+
+use eth2::lighthouse::SyncCommitteeReward;
+use state_processing::per_block_processing::altair::sync_committee::compute_sync_aggregate_rewards;
+use types::{BeaconBlockRef, BeaconState, ExecPayload};
+use std::collections::HashMap;
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_sync_committee_rewards<Payload: ExecPayload<T::EthSpec>>(
+        &self,
+        block: BeaconBlockRef<'_, T::EthSpec, Payload>,
+        state: &mut BeaconState<T::EthSpec>,
+    ) -> Result<Vec<SyncCommitteeReward>, BeaconChainError> {
+        if block.slot() != state.slot() {
+            return Err(BeaconChainError::BlockRewardSlotError);
+        }
+
+        let spec = &self.spec;
```

Comment:
We need to build the committee cache here in case it isn't already built. Otherwise we'll get an error when we compute rewards at `slot % 32 == 1` and load an epoch boundary state without any block replay (the block replayer will build the cache in all other cases).

```rust
state.build_committee_cache(RelativeEpoch::Current, spec)?;
```